Looms.PoS/Looms.PoS.Application/Features/OrderItem/Commands/CreateOrderItem/CreateOrderItemsRequestValidator.cs

```diff
@@ -0,0 +1,85 @@
+using FluentValidation;
+using Looms.PoS.Application.Models.Requests;
+using Looms.PoS.Application.Utilities.Validators;
+using Looms.PoS.Domain.Enums;
+using Looms.PoS.Domain.Interfaces;
+
+namespace Looms.PoS.Application.Features.Discount.Commands.CreateDiscount;
+
+public class CreateOrderItemRequestValidator : AbstractValidator<CreateOrderItemRequest>
+{
+    public CreateOrderItemRequestValidator(
+        IOrdersRepository ordersRepository, 
+        IDiscountsRepository discountsRepository,
+        IProductVariationRepository variationRepository,
+        IProductsRepository productsRepository)
+    {
+        RuleLevelCascadeMode = CascadeMode.Stop;
+
+        RuleFor(x => x)
+            .Must(x => x.ServiceId != null || x.ProductId != null)
+            .WithMessage("ServiceId or ProductId is required.");
+
+// TODO: add rules for service ids with custom async validation like from GetOrderItemsQueryValidator
+
+        RuleFor(x => x.ProductId!)
+            .MustBeValidGuid()
+            .CustomAsync(async (productId, context, cancellationToken) => 
+            {
+                var product = await productsRepository.GetAsync(Guid.Parse(productId));
+                if (product is null)
+                {
+                    context.AddFailure("Product not found.");
+                }else{
+                    if(product!.Quantity < context.InstanceToValidate.Quantity)
+                    {
+                        context.AddFailure("Product quantity is too low.");
+                    }
+                }
+
+            })
+            .When(x => x.ServiceId is null && x.ProductId != null);
+
+        RuleFor(x => x.ProductVariationId!)
+            .MustBeValidGuid()
+            .CustomAsync(async (variationId, context, cancellationToken) => 
+            {
+                var variation = await variationRepository.GetAsync(Guid.Parse(variationId));
+                if (variation is null)
+                {
+                    context.AddFailure("Variation not found.");
+
+                    if(variation!.Quantity < context.InstanceToValidate.Quantity)
+                    {
+                        context.AddFailure("Variation quantity is too low.");
+                    }
+                }
+            })
+            .When(x => x.ServiceId is null && x.ProductId != null && x.ProductVariationId != null);
+
+        RuleFor(x => x.DiscountId!)
+            .MustBeValidGuid()
+            .CustomAsync(async (discountId, context, cancellationToken) =>
+            {
+                var discount = await discountsRepository.GetAsync(Guid.Parse(discountId));
+                if (discount is null)
+                {
+                    context.AddFailure("Discount not found.");
```

Comment:
This will return 400 should be 404, repository should throw exception if it doesnt

Reply:
removed from all validators